test/toolbox_image.sh

```diff
@@ -254,36 +263,124 @@ test_system_update_entrypoint()
+{
+    new_boot_start="2047"
+    new_boot_size="$((ROOTFS_START - new_boot_start))"
+
+    # In every line in the partition table look for a string "p1<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s/p1.*type/p1 : start=     ${new_boot_start}, size=     ${new_boot_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 0
+}
+
+test_execute_disk_prepare_grow_beyond_disk_end_nok()
+{
+    new_userdata_size="$((STORAGE_DEVICE_END - ROOTFS_START))"
+
+    # In every line in the partition table look for a string "p3<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s/p3.*type/p3 : start=     ${USERDATA_START}, size=     ${new_userdata_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 0
+}
+
+test_execute_resize_partition_grow_rootfs_ok()
+{
+    userdata_size="$((STORAGE_DEVICE_SIZE - USERDATA_START))"
+    new_userdata_size="$((userdata_size / 2))"
+    new_userdata_start="$((USERDATA_START + new_userdata_size))"
+    new_rootfs_size="$((new_userdata_start - ROOTFS_START))"
+    # In every line in the partition table look for a string "p3<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s/p2.*type/p2 : start=       ${ROOTFS_START}, size=     ${new_rootfs_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+    sed -i "s/p3.*type/p3 : start=     ${new_userdata_start}, size=     ${new_userdata_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 1
+    helper_test_disk_integrity || return 1
+}
+
+test_execute_disk_prepare_grow_boot_ok()
+{
+    rootfs_size="$((USERDATA_START - ROOTFS_START))"
+    new_rootfs_size="$((rootfs_size / 2))"
+    new_rootfs_start="$((ROOTFS_START + new_rootfs_size))"
+    new_boot_size="$((new_rootfs_start + BOOT_START))"
+
+    # In every line in the partition table look for a string "p2<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s/p1.*type/p1 : start=     ${BOOT_START}, size=      ${new_boot_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+    sed -i "s/p2.*type/p2 : start=     ${new_rootfs_start}, size=      ${new_rootfs_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 1
+    helper_test_disk_integrity || return 1
+  }
+
+test_execute_disk_prepare_sha512_nok()
+{
+    partition_table_file="/tmp/partition_table"
+    sfdisk -d "${rootfs_dir}${STORAGE_DEVICE_IMG}" > "${rootfs_dir}${partition_table_file}"
+    chroot "${rootfs_dir}" sha512sum "${PARTITION_TABLE_FILE}" > "${rootfs_dir}${PARTITION_TABLE_FILE}.sha512"
+    echo "some extra data" >> "${rootfs_dir}${partition_table_file}"
```

Comment:
note, that this test will fail because the 'extra data' will not validate, not because the shasum is invalid.

so i'd say, rewrite ($RANDOM) character(s) in the sha15sum?

Reply:
No im am making the partition file corrupt so that the md512sum does not validate anymore.